sdk/python/feast/repo_operations.py

```diff
@@ -108,6 +109,17 @@ def parse_repo(repo_root: Path) -> ParsedRepo:
     return res
 
 
+def assert_table_exists(table_ref):
```

Comment:
Let's move this to the data source and make it part of the public API:

1. Add `validate()` method to `DataSource` class that raises `NotImplementedError`.
2. Override the method in `BigQuerySource` and move the contents of this function to that method
3. Override the method in `FileSource` without doing any validations there for now (we can validate that the file exists locally, in s3, or in gs in a separate PR)

Reply:
Also, let's move the conditional check from below (`not data_source.query`) here.